File: backend/bill-service/src/main/java/com/easyerp/billservice/requests/BillRequest.java
package com.easyerp.billservice.requests;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@JsonIgnoreProperties(ignoreUnknown = true)
public class BillRequest {
    private Long clientId;
    private Double tva;
    private List<BillLineRequest> lines;
    private boolean draft;
    private Long quoteId;
}

File: client/src/models/ApiRequests/BillRequest.ts
import { BillLineRequest } from "./BillLineRequest";

export class BillRequest {
  clientId?: number;
  tva: number;
  draft: boolean;
  lines: BillLineRequest[];

  constructor() {
    this.tva = 0;
    this.draft = true;
    this.lines = [];
  }
}
